### PR TITLE
Feature/add-to-cart

### PR DESCRIPTION
**Pull request by fredbordel**
No changed made on backend
Folder involved in changes: client/public, client/component, /hooks, /reducers
Files modified : Feed.jsx, Cart.jsx, dataReducer.jsx, App.jsx, useApplicationData.jsx
File created: Cart.scss

Feed -> Added logic
Cart -> Renders the cart items
dataReducer -> added SET_CART state
App -> added route for Cart and passing it props
useApplicationData -> Added function (setCart) that updated cart state with dispatch

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,54 +1,69 @@
-import React from 'react';
-import { 
-  BrowserRouter as Router, 
-  Switch,
-  Route
-} from "react-router-dom";
-import Landing from './components/Landing';
-import Dashboard from './components/Dashboard';
+import React from "react";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
```
Comment (Miaouchkinz) [inline client/src/App.jsx]: You prefer this way over this?

```
import { 
  BrowserRouter as Router, 
  Switch,
  Route
} from "react-router-dom";
```

File: client/src/components/Feed.jsx [hunk 2]
```diff
@@ -1,14 +1,32 @@
 import React, { useState } from "react";
+import { Link } from "react-router-dom";
```
Comment (Miaouchkinz) [inline client/src/components/Feed.jsx]: I wonder if we can pass that done from app In the same way that we pass down router props? hrmmm! Maybe something to tackle later for refactoring 🤔

File: client/src/components/Feed.jsx [hunk 3]
```diff
@@ -41,18 +61,42 @@ export default function Feed({ clothing, clothingCategories }) {
...
+  ) {
+    if (
+      addToCartCount[0].totalCount === 0 &&
+      !addToCartCount.some(key => key[clothingId] === clothingId)
+    ) {
+      setaddToCartCount([
+        ...addToCartCount,
+        { totalCount: 1, [clothingId]: clothingId }
+      ]);
+      callback(clothingId, clothingSize, clothingCategory);
```
Comment (Miaouchkinz) [inline client/src/components/Feed.jsx]: What does this callback do?

File: client/src/components/Header.jsx [hunk 4]
```diff
@@ -2,6 +2,9 @@ import React from "react";
 
 export default function Header(props) {
   return (
-    <h1>This is Header</h1>
-  )
-}
\ No newline at end of file
+    <div>
+      <span></span>
```
Comment (Miaouchkinz) [inline client/src/components/Header.jsx]: What do the span do? Just placeholder for now i imagine?

File: client/src/components/Untitled-1 [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+return result
+};
+
+
+firstFunction(5, 6, 7
+
+
+
+const famille = function(param1, param2, param3) {
+  return `Ma famille est composée de ${param1}, ${param2}, ${param3}`
```
Comment (Miaouchkinz) [inline client/src/components/Untitled-1]: hahaha

Did you mean to push this file? :P

File: client/src/hooks/useApplicationData.jsx [hunk 6]
```diff
@@ -14,72 +15,92 @@ const useApplicationData = () => {
...
   useEffect(() => {
     Promise.all([
-      axios.get('http://localhost:3001/api/users'),
-      axios.get('http://localhost:3001/logged_in', { withCredentials: true }),
-      axios.get('/api/clothings'),
-      axios.get('/api/clothing_categories')
+      axios.get("http://localhost:3001/api/users"),
+      axios.get("http://localhost:3001/logged_in", { withCredentials: true }),
+      axios.get("http://localhost:3001/api/clothings"),
+      axios.get("http://localhost:3001/api/clothing_categories")
```
Comment (Miaouchkinz) [inline client/src/hooks/useApplicationData.jsx]: I think francis was saying we should take out the "http://localhost:3001" for all of them!

File: client/src/App.jsx (hunk 1 above)
Comment (fredbordel) [reply client/src/App.jsx]: I think it may be a prettier thing because I don't have an opinion on this ahahah.

Comment (fredbordel) [reply client/src/App.jsx]: I don't remember changing it on purpose

File: client/src/components/Feed.jsx (hunk 2 above)
Comment (fredbordel) [reply client/src/components/Feed.jsx]: Yes maybe passing Link at the same place than Browser, switch, etc and then passing Link as props you mean?

File: client/src/components/Feed.jsx (hunk 3 above)
Comment (fredbordel) [reply client/src/components/Feed.jsx]: This callback is the "place holder" for the setCart function, that I pass as an argument when calling addToCartButtonClicked at line 143

File: client/src/components/Header.jsx (hunk 4 above)
Comment (fredbordel) [reply client/src/components/Header.jsx]: woops lolll no it does nothing for now. If we decided to use the Header component, we'll probably modify that a little bit

File: client/src/components/Untitled-1 (hunk 5 above)
Comment (fredbordel) [reply client/src/components/Untitled-1]: AHAHAHHAHA NOOOO omg this was for Laurie, I did not know I had created that!!! I'll delete it now

File: client/src/hooks/useApplicationData.jsx (hunk 6 above)
Comment (fredbordel) [reply client/src/hooks/useApplicationData.jsx]: True, I'm changing it right now

File: client/src/components/Feed.jsx (hunk 2 above)
Comment (Miaouchkinz) [reply client/src/components/Feed.jsx]: yeah exactly, but let's save that for later! Not a big deal right now :)

File: client/src/components/Feed.jsx (hunk 3 above)
Comment (Miaouchkinz) [reply client/src/components/Feed.jsx]: @fredbordel Can we give it a name so it's clearer what it does?

File: client/src/components/Untitled-1 (hunk 5 above)
Comment (Miaouchkinz) [reply client/src/components/Untitled-1]: hahahaha ok!!!!

**Pull request merged**